include/nbl/builtin/hlsl/subgroup/fft.hlsl

```diff
@@ -12,94 +12,6 @@ namespace hlsl
-        hi.imag(exchanged.y);
-    }
-    // Get twiddle with k = subgroupInvocation mod stride, N = 2 * stride
-    fft::DIF<Scalar>::radix2(fft::twiddle<false, Scalar>(glsl::gl_SubgroupInvocationID() & (stride - 1), stride << 1), lo, hi);   
-}
-
-template<typename Scalar>
-void FFT<false, Scalar>(NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi) 
-{
-    const uint32_t subgroupSize = glsl::gl_SubgroupSize();  //This is N/2
-    const uint32_t doubleSubgroupSize = subgroupSize << 1;  //This is N
-    
-    // special first iteration
-    fft::DIF<Scalar>::radix2(fft::twiddle<false, Scalar>(glsl::gl_SubgroupInvocationID(), doubleSubgroupSize), lo, hi);                                                                                   
-        
-    // Decimation in Frequency
-    for (uint32_t stride = subgroupSize >> 1; stride > 0; stride >>= 1)
-        FFT_loop<false, Scalar>(stride, lo, hi);
-}   
-
-
-// ---------------------------------------- Inverse transform - DIT -------------------------------------------------------
-template<typename Scalar>
-void FFT_loop<true, Scalar>(uint32_t stride, NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi) 
-{
-    // Get twiddle with k = subgroupInvocation mod stride, N = 2 * stride
-    fft::DIT<Scalar>::radix2(fft::twiddle<true, Scalar>(glsl::gl_SubgroupInvocationID() & (stride - 1), stride << 1), lo, hi);   
-
-    const bool topHalf = (glsl::gl_SubgroupInvocationID() & stride) != 0;
-    const vector <Scalar, 2> toTrade = topHalf ? vector <Scalar, 2>(lo.real(), lo.imag()) : vector <Scalar, 2>(hi.real(), hi.imag());
-    const vector <Scalar, 2> exchanged = glsl::subgroupShuffleXor< vector <Scalar, 2> > (toTrade, stride);
-    if (topHalf)
-    {
-        lo.real(exchanged.x);
-        lo.imag(exchanged.y);
-    }
-    else
-    {
-        hi.real(exchanged.x);
-        hi.imag(exchanged.y);
-    }
-    
-}
-
-template<typename Scalar>
-void FFT<true, Scalar>(NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi) 
-{
-    const uint32_t subgroupSize = glsl::gl_SubgroupSize();  //This is N/2
-    const uint32_t doubleSubgroupSize = subgroupSize << 1;  //This is N                                                                           
-        
-    // Decimation in Time
-    for (uint32_t stride = 1; stride < subgroupSize; stride <<= 1)
-        FFT_loop<Scalar, true>(stride, lo, hi);
-        
-    // special last iteration 
-    fft::DIT<Scalar>::radix2(fft::twiddle<Scalar, true>(glsl::gl_SubgroupInvocationID(), doubleSubgroupSize), lo, hi);
-    divides_assign< complex_t<Scalar> > divAss;
-    divAss(lo, doubleSubgroupSize);
-    divAss(hi, doubleSubgroupSize);            
-}   
-
-*/
-
 // -----------------------------------------------------------------------------------------------------------------------------------------------------------------
 template<uint16_t K, bool Inverse, typename Scalar, class device_capabilities=void>
```

Comment:
don't call it `K` call it `Radix` cause otherwise people can get confused what the `K` is

Reply:
also move it to AFTER `device_capabilities`, and default it to `2`

Otherwise user has to specify its `radix=2` every time, which they probably don't want (only inverse vs. forward)

Reply:
btw you don't use the `device_capabilities` at all, so whats the point of taking them?

Reply:
ah ok you can use `device_capabilities` after #587 gets merged to `[unroll(device_traits<device_capabilities>::subgroupMinSize)]` your for loops

Reply:
device traits PR is now merged to master